### DISCUSSION TITLE
I couldn't install with composer

### DISCUSSION THREAD

**Discussion by maghraba** [PHPOffice/PhpSpreadsheet#3135] [General]
Composer is installed in the Windows environment.

PhP/SpeadSheet downloaded with Github Desktop.

I start a cmd from the PhP/SpeadSheet folder.

"composer install" and "composer require PhP/SpeadSheet" throw the same thing: "Root package 'phpoffice/phpspreadsheet' cannot require itself in its composer.json".

What and where should be rewritten or done differently?

D:\xampp\htdocs\tablazatok\PhpSpreadsheet-master>composer require phpoffice/phpspreadsheet
PHP:  syntax error, unexpected BOOL_FALSE in D:\xampp\php\windowsXamppPhp\php.ini on line 1083
PHP Warning:  PHP Startup: Unable to load dynamic library 'php_sqlsrv_81_ts_x86' (tried: ext\php_sqlsrv_81_ts_x86 (%1: nem Win32 alkalmaz├ís), ext\php_php_sqlsrv_81_ts_x86.dll (A megadott modul nem tal├ílhat├│)) in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library 'php_sqlsrv_81_ts_x86' (tried: ext\php_sqlsrv_81_ts_x86 (%1: nem Win32 alkalmaz├ís), ext\php_php_sqlsrv_81_ts_x86.dll (A megadott modul nem tal├ílhat├│)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'php_pdo_sqlsrv_81_ts_x86' (tried: ext\php_pdo_sqlsrv_81_ts_x86 (%1: nem Win32 alkalmaz├ís), ext\php_php_pdo_sqlsrv_81_ts_x86.dll (A megadott modul nem tal├ílhat├│)) in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library 'php_pdo_sqlsrv_81_ts_x86' (tried: ext\php_pdo_sqlsrv_81_ts_x86 (%1: nem Win32 alkalmaz├ís), ext\php_php_pdo_sqlsrv_81_ts_x86.dll (A megadott modul nem tal├ílhat├│)) in Unknown on line 0
PHP Warning:  syntax error, unexpected BOOL_FALSE in Unknown on line 1083
 in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php on line 390
PHP Stack trace:
PHP   1. {main}() C:\ProgramData\ComposerSetup\bin\composer.phar:0
PHP   2. require() C:\ProgramData\ComposerSetup\bin\composer.phar:29
PHP   3. Composer\XdebugHandler\XdebugHandler->check() phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:24
PHP   4. Composer\XdebugHandler\XdebugHandler->prepareRestart() phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:147
PHP   5. Composer\XdebugHandler\XdebugHandler->writeTmpIni($iniFiles = [0 => 'D:\\xampp\\php\\windowsXamppPhp\\php.ini'], $tmpDir = 'C:\\Users\\gmagy\\AppData\\Local\\Temp', $error = NULL) phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:341
PHP   6. parse_ini_string($ini_string = '[PHP]\r\n\r\n;;;;;;;;;;;;;;;;;;;\r\n; About php.ini   ;\r\n;;;;;;;;;;;;;;;;;;;\r\n; PHP\'s initialization file, generally called php.ini, is responsible for\r\n; configuring many of the aspects of PHP\'s behavior.\r\n\r\n; PHP attempts to find and load this configuration from a number of locations.\r\n; The following is a summary of its search order:\r\n; 1. SAPI module specific location.\r\n; 2. The PHPRC environment variable. (As of PHP 5.2.0)\r\n; 3. A number of predefined registry keys on Windows (As of') phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:390

Warning: syntax error, unexpected BOOL_FALSE in Unknown on line 1083
 in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php on line 390

Call Stack:
    0.0275    4921232   1. {main}() C:\ProgramData\ComposerSetup\bin\composer.phar:0
    0.0280    4926024   2. require('phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer') C:\ProgramData\ComposerSetup\bin\composer.phar:29
    0.0360    5264088   3. Composer\XdebugHandler\XdebugHandler->check() phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:24
    0.0360    5264464   4. Composer\XdebugHandler\XdebugHandler->prepareRestart() phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:147
    0.0362    5265072   5. Composer\XdebugHandler\XdebugHandler->writeTmpIni($iniFiles = [0 => 'D:\\xampp\\php\\windowsXamppPhp\\php.ini'], $tmpDir = 'C:\\Users\\gmagy\\AppData\\Local\\Temp', $error = NULL) phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:341
    0.0421    5454296   6. parse_ini_string($ini_string = '[PHP]\r\n\r\n;;;;;;;;;;;;;;;;;;;\r\n; About php.ini   ;\r\n;;;;;;;;;;;;;;;;;;;\r\n; PHP\'s initialization file, generally called php.ini, is responsible for\r\n; configuring many of the aspects of PHP\'s behavior.\r\n\r\n; PHP attempts to find and load this configuration from a number of locations.\r\n; The following is a summary of its search order:\r\n; 1. SAPI module specific location.\r\n; 2. The PHPRC environment variable. (As of PHP 5.2.0)\r\n; 3. A number of predefined registry keys on Windows (As of') phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/composer/xdebug-handler/src/XdebugHandler.php:390

Composer is operating slower than normal because you have Xdebug enabled. See https://getcomposer.org/xdebug
Using version ^1.25 for phpoffice/phpspreadsheet
Root package 'phpoffice/phpspreadsheet' cannot require itself in its composer.json

**Comment by MarkBaker** [comment]
You should run the `composer require phpoffice/phpspreadsheet` command from the root folder of your own project, and that will download PhpSpreadsheet and all its dependencies for you; you shouldn't download PhpSpreadsheet manually.
Alternatively, if you already have a `composer.json` file in your own project; add PhpSpreadsheet as a  `require` entry in that json, and run `composer install`.

Either user composer to do the installation; or download and set it all up manually; but not a combination of the two

It helps to understand what [composer](https://code.tutsplus.com/tutorials/what-is-composer-for-php-and-how-to-install-it--cms-35160) is and does for you

**Reply by maghraba** [reply]
Thanks for the reply!
A few more questions remain.
-Which is OWN library?
-After it is installed, how do I use it?
For example, if there is a trysheet project in the htdocs directory, should I put a composer.json file in its root and then run the composer require ... command?
